pi5/board.go

```diff
@@ -162,16 +152,109 @@ func (b *pinctrlpi5) Reconfigure(
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
+	// make sure every pin has a name. We already know every pin has a pin
+	for _, c := range newConf.Pins {
+		if c.Name == "" {
+			c.Name = c.Pin
```

Comment:
I'm trying to figure out if overwriting the config is an antipattern.

If we overwrite the config we've received from app, and a new config is sent from app while we're overwriting the old one, wouldn't;t that cause ambiguity? I don't think we're locking anything about the config object, nor should we.

Reply:
I mean I can ask Cheuk if he thinks this would cause some weird issue but I don't think it should. I could add a few more lines of code to make a different pin struct instead of using newConf

Reply:
Let's check with cheuk! He'll definitely give us the solid truth.

Reply:
from a technical sense, this will be fine. especially since this is happening on a module, the config here that is being changed is not used by the viam-server to figure out if it should reconfigure. if this would cause issues, I would expect unnecessary reconfigurations even if the config has not changed. I'm guessing you already did some testing on this, so if you didn't run into any issues I don't think there are any issues.

I prefer not changing the config as part of reconfiguration and just using defaults at the point where the blank field would be used, but I think that's more of a matter of personal preference

Reply:
Feel free to keep as is, then John. You can add a TODO above the code to clean up the pin naming by using the pin number in data.go or the pigpio map in the pi4 to call out that we're aware this is messy and needs to be refactored at a future date.

Reply:
i don't follow what you mean by this/don't understand why this particular piece of code is messy. 

addUserDefinedNames checks that all of the pins are actually pins, so I can make sure the comments call that out

Reply:
I would prefer the final form of this driver to have a pin mapping that stores everything about the pin (number, given name from config, debounce timer, mode), right now there are multiple sources of truth in naming the pins as well as setting pin behaviours from configuration and APIs. The cleanest version of both these drivers would call the same functions in both. But the great code hygiene of this module is not a priority for me at this point.